resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -381,4 +382,21 @@ object KubernetesUtils extends Logging {
       }
     }
   }
+
+  @Since("3.3.0")
+  def createOrReplaceResource(
```

Comment:
Given the function semantic is destructive (destroying the owner pod), `createOrReplaceResource` looks inadequite to me. We should have more explicit name which warns the pod deletion.